osgp/platform/osgp-adapter-domain-core/src/test/java/org/opensmartgridplatform/adapter/domain/core/application/services/DeviceManagementServiceTest.java

```diff
@@ -0,0 +1,217 @@
+import org.opensmartgridplatform.domain.core.valueobjects.CdmaSettings;
+import org.opensmartgridplatform.domain.core.valueobjects.Certification;
+import org.opensmartgridplatform.domain.core.valueobjects.DeviceLifecycleStatus;
+import org.opensmartgridplatform.domain.core.valueobjects.EventNotificationType;
+import org.opensmartgridplatform.dto.valueobjects.EventNotificationMessageDataContainerDto;
+import org.opensmartgridplatform.shared.exceptionhandling.ComponentType;
+import org.opensmartgridplatform.shared.exceptionhandling.FunctionalException;
+import org.opensmartgridplatform.shared.infra.jms.RequestMessage;
+import org.opensmartgridplatform.shared.infra.jms.ResponseMessage;
+import org.opensmartgridplatform.shared.infra.jms.ResponseMessageResultType;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class DeviceManagementServiceTest {
+	
+	@Mock
+	private TransactionalDeviceService transactionalDeviceService;
+	
+	@Mock
+	private DeviceDomainService deviceDomainService;
+	
+	@Mock
+	private DomainCoreMapper domainCoreMapper;
+	
+	@Mock
+	private OsgpCoreRequestMessageSender osgpCoreRequestManager;
+	
+	@Mock
+	private WebServiceResponseMessageSender webServiceResponseMessageSender;
+
+	@Mock
+	private OrganisationDomainService organisationDomainService;
+	
+	@InjectMocks
+	private DeviceManagementService deviceManagementService;
+
+	@Captor
+	private ArgumentCaptor<RequestMessage> argumentReqM;
+
+	@Captor
+	private ArgumentCaptor<ResponseMessage> argumentResM;
+
+	@Captor
+	private ArgumentCaptor<String> argumentStringOne;
+
+	@Captor
+	private ArgumentCaptor<String> argumentStringTwo;
+
+	@Captor
+	private ArgumentCaptor<Integer> argumentInt;
+
+	@BeforeEach
+	public void init() throws UnknownEntityException {
+		when(this.organisationDomainService.searchOrganisation(any())).thenReturn(new Organisation());
+	}
+	
+	@Test
+	public void testSetEventNotifications() throws FunctionalException {
+		final List<EventNotificationType> eventNotifications = Arrays.asList(
+				EventNotificationType.COMM_EVENTS,
+				EventNotificationType.DIAG_EVENTS);
+		final Device device = mock(Device.class);
+		when(device.getIpAddress()).thenReturn("testIp");
+		when(this.deviceDomainService.searchActiveDevice("testDevice", ComponentType.DOMAIN_CORE)).thenReturn(device);
+		
+		this.deviceManagementService.setEventNotifications("testOrganisation", "testDevice", "testUid",
+				eventNotifications, "testMessageType", 1);
+
+		verify(this.osgpCoreRequestManager).send(this.argumentReqM.capture(), this.argumentStringOne.capture(),
+				this.argumentInt.capture(), this.argumentStringTwo.capture());
+
+		final RequestMessage expectedRM = new RequestMessage("testUid", "testOrganisation", "testDevice",
+				new EventNotificationMessageDataContainerDto(this.domainCoreMapper.mapAsList(eventNotifications,
+						org.opensmartgridplatform.dto.valueobjects.EventNotificationTypeDto.class)));
```

Comment:
- rename expectedRM to expectedRequestMessage (also in other tests)
- refactor the creation of a request message to a method, so it can be used in other tests as well